ts/Core/Series/Series.ts

```diff
@@ -4961,7 +4969,15 @@ class Series {
                     chart.sharedClips[sharedClipKey] = clipRect.destroy();
                 }
                 if (markerClipRect) {
-                    chart.sharedClips[sharedClipKey + 'm'] = markerClipRect.destroy();
+                    chart.sharedClips[sharedClipKey + 'm'] =
+                        markerClipRect.destroy();
+                }
+
+                // Remove the temporary clipping rectangle when the series
+                // animation has finished after chart.addSeries(), (#4406)
+                const tempClip = tempClipKey && chart.sharedClips[tempClipKey];
+                if (tempClipKey && tempClip) {
+                    chart.sharedClips[tempClipKey] = tempClip.destroy();
```

Comment:
I think this should be removed since the existing `clipRect.destroy` is going to deal with it for the non-stock case (no `clipBox`) and then add a `else if (isTemp)` to it that chops off the temp prefix and (since shared clips are not destroyed in stock):

- If a shared clip with that key already exists, changes the group clip to that and destroys the temp one.
- If not, keeps the temp one and moves it to the non-temp key and updates `this.sharedClipKey`.

This should fix the visual diff.